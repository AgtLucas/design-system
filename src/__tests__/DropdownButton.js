import React from 'react'
import renderer from 'react-test-renderer'
import { DropdownButton, ThemeProvider } from '..'

describe('DropdownButton', () => {
  test('renders without any prop', () => {
    const json = renderer
      .create(
        <ThemeProvider>
          <DropdownButton>Dropdown Button</DropdownButton>
        </ThemeProvider>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with "open" props', () => {
    const json = renderer
      .create(
        <ThemeProvider>
          <DropdownButton open>Dropdown Button</DropdownButton>
        </ThemeProvider>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
