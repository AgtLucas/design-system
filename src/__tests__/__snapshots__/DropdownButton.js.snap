// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownButton renders with "open" props 1`] = `
.c2 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c2:disabled {
  opacity: 0.25;
}

.c2:hover {
  background-color: #049;
}

.c1 {
  background-color: transparent;
  border-width: 2px;
  border-style: solid;
  color: #007aff;
  border-color: #007aff;
  cursor: pointer;
  width: 100%;
  height: 100%;
  min-height: 44px;
  border-color: #d1d6db;
}

.c1:hover {
  color: #049;
  border-color: #049;
  background-color: transparent;
}

.c1:hover {
  border-color: #007aff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <button
    aria-controls="dropdownMenu"
    aria-expanded="true"
    aria-haspopup="true"
    className="c1 c2"
    id="dropdownButton"
    open={true}
    type="button"
  >
    Dropdown Button
  </button>
</div>
`;

exports[`DropdownButton renders without any prop 1`] = `
.c2 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c2:disabled {
  opacity: 0.25;
}

.c2:hover {
  background-color: #049;
}

.c1 {
  background-color: transparent;
  border-width: 2px;
  border-style: solid;
  color: #007aff;
  border-color: #007aff;
  cursor: pointer;
  width: 100%;
  height: 100%;
  min-height: 44px;
  border-color: #d1d6db;
}

.c1:hover {
  color: #049;
  border-color: #049;
  background-color: transparent;
}

.c1:hover {
  border-color: #007aff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <button
    aria-controls="dropdownMenu"
    aria-expanded="false"
    aria-haspopup="true"
    className="c1 c2"
    id="dropdownButton"
    open={false}
    type="button"
  >
    Dropdown Button
  </button>
</div>
`;
